src/sql/workbench/contrib/extensions/electron-browser/extensionsActions.ts

```diff
@@ -0,0 +1,86 @@
+import { IExtensionRecommendation, IExtensionManagementServerService } from 'vs/platform/extensionManagement/common/extensionManagement';
+import { INotificationService, Severity } from 'vs/platform/notification/common/notification';
+import { IOpenerService } from 'vs/platform/opener/common/opener';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { PagedModel } from 'vs/base/common/paging';
+import { IConfigurationService, ConfigurationTarget } from 'vs/platform/configuration/common/configuration';
+
+export class ShowVisualizerExtensionsAction extends Action {
+
+	static readonly ID = 'workbench.extensions.action.showVisualizerExtensions';
+	static LABEL = localize('showVisualizerExtensions', "Show Visualizers");
+
+	constructor(
+		id: string,
+		label: string,
+		@IViewletService private readonly viewletService: IViewletService
+	) {
+		super(id, label, undefined, true);
+	}
+
+	run(): Promise<void> {
+		return this.viewletService.openViewlet(VIEWLET_ID, true)
+			.then(viewlet => viewlet as IExtensionsViewlet)
+			.then(viewlet => {
+				viewlet.search('@visualizerExtensions');
+				viewlet.focus();
+			});
+	}
+}
+
+export class InstallVisualizerExtensionsAction extends Action {
+
+	static readonly ID = 'workbench.extensions.action.installVisualizerExtensions';
+	static LABEL = localize('installVisualizerExtensions', "Install Visualizer Extensions");
+
+	private _recommendations: IExtensionRecommendation[] = [];
+	get recommendations(): IExtensionRecommendation[] { return this._recommendations; }
+	set recommendations(recommendations: IExtensionRecommendation[]) { this._recommendations = recommendations; this.enabled = this._recommendations.length > 0; }
+
+	constructor(
+		id: string = InstallVisualizerExtensionsAction.ID,
+		label: string = InstallVisualizerExtensionsAction.LABEL,
+		recommendations: IExtensionRecommendation[],
+		@IViewletService private readonly viewletService: IViewletService,
+		@INotificationService private readonly notificationService: INotificationService,
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IOpenerService private readonly openerService: IOpenerService,
+		@IExtensionsWorkbenchService private readonly extensionWorkbenchService: IExtensionsWorkbenchService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IExtensionManagementServerService private readonly extensionManagementServerService: IExtensionManagementServerService
+	) {
+		super(id, label, 'extension-action');
+		this.recommendations = recommendations;
+	}
+
+	run(): Promise<any> {
+		return this.viewletService.openViewlet(VIEWLET_ID, true)
+			.then(viewlet => viewlet as IExtensionsViewlet)
+			.then(viewlet => {
+				viewlet.search('@visualizerExtensions');
+				viewlet.focus();
+				const names = this.recommendations.map(({ extensionId }) => extensionId);
```

Comment:
Let's cover case where this.recommendations is null since it's got to be set.

Reply:
Thisclass, InstallVisualizerExtensionsAction, is only called when there are extensions to be recommended. This logic is handled in extensionTipsServices, and InstallVisualizerExtensionsAction is called in promptVisualizerExtensions(). 

Is the null case still necessary?

Reply:
General principle is to be self-contained. E.g. in this class you have a value that's not set in the constructor, but we "expected" to be there. Given this we would normally do null checking. I'm OK if you decide not to, it's more about defense in depth.